setup.py

```diff
@@ -0,0 +1,132 @@
+"""
+setuptools-based build of ROOT.
+
+This script uses setuptools API to steer a custom CMake build of ROOT. All build
+configuration options are specified in the class responsible for building. A
+custom extension module is injected in the setuptools setup to properly generate
+the wheel with CPython extension metadata. Note that ROOT is first installed via
+CMake into a temporary directory, then the ROOT installation artifacts are moved
+to the final Python environment installation path, which often starts at
+${ENV_PREFIX}/lib/pythonXX.YY/site-packages, before being packaged as a wheel.
+"""
+
+import os
+import pathlib
+import shlex
+import subprocess
+import tempfile
+
+from setuptools import Extension, find_packages, setup
+from setuptools.command.build import build as _build
+from setuptools.command.install import install as _install
+
+# Get the long description from the README file
+SOURCE_DIR = pathlib.Path(__file__).parent.resolve()
+LONG_DESCRIPTION = (SOURCE_DIR / "README.md").read_text(encoding="utf-8")
+
+BUILD_DIR = tempfile.mkdtemp()
+INSTALL_DIR = tempfile.mkdtemp()
+
+# Name given to an internal directory within the build directory
+# used to mimick the structure of the target installation directory
+# in the user Python environment, usually named "site-packages"
+ROOT_BUILD_INTERNAL_DIRNAME = "mock_site_packages"
+
+
+class ROOTBuild(_build):
+    def run(self):
+        _build.run(self)
+
+        # Configure ROOT build
+        configure_command = shlex.split(
+            "cmake -GNinja -Dccache=ON "
```

Comment:
```suggestion
            "cmake "
```
Now that we have a `setup.py`, people will come and call `pip install root_src/`. Is it really necessary to use `Ninja` and `ccache` in general? These packages are probably not installed anywhere.

I'm also fine keeping this, but it's just something to consider.